src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts

```diff
@@ -113,25 +112,42 @@ export class PowerPagesChatParticipant {
             };
         }
 
-        const userPrompt = request.prompt;
+        if (request.command) {
+            //TODO: Handle command scenarios
 
-        if (!userPrompt) {
+        } else {
 
-            //TODO: Show start message
+            const userPrompt = request.prompt;
 
-            return {
-                metadata: {
-                    command: ''
-                }
-            };
-        }
+            if (!userPrompt) {
 
-        //TODO: Handle form and list scenarios
-        const llmResponse = await sendApiRequest([{ displayText: userPrompt, code: '' }], { dataverseEntity: '', entityField: '', fieldType: '' }, this.orgID, intelligenceApiToken, '', '', [], this.telemetry, intelligenceEndpoint, geoName);
+                //TODO: String approval is required
+                stream.markdown(NO_PROMPT_MESSAGE);
 
-        stream.markdown(llmResponse[0].displayText);
+                return {
+                    metadata: {
+                        command: ''
+                    }
+                };
+            }
+
+            const {activeFileParams} = getActiveEditorContent();
+
+            const componentInfo = await getComponentInfo(this.telemetry, this.orgUrl, activeFileParams);
+
+            const llmResponse = await sendApiRequest([{ displayText: userPrompt, code: '' }], activeFileParams, this.orgID, intelligenceApiToken, '', '', componentInfo, this.telemetry, intelligenceEndpoint, geoName);
 
-        stream.markdown('\n```typescript\n' + llmResponse[0].code + '\n```');
+            llmResponse.forEach((response: { displayText: string | vscode.MarkdownString; code: string; }) => {
+                if (response.displayText) {
+                    stream.markdown(response.displayText);
+                }
+                if (response.code) {
+                    stream.markdown('\n```javascript\n' + response.code + '\n```');
```

Comment:
how did we finally fix this markdown rendering?

Reply:
Adding the newline character worked for our responses. The example I was trying must be missing something on same lines.